File: accanalysis.R
rm(list=ls())
library(ggplot2)
library(gridExtra)

fname <- '/Users/mviana/Desktop/deep-learning-models/log/20170626T103649'

Table <- data.frame(read.table(paste(fname,'.csv',sep=''), header=T, sep=','))

Table$crossval <- factor(Table$crossval)

f1 <- ggplot(Table) + geom_line(aes(epoch,train_loss,group=crossval,col=crossval), alpha=0.5) + theme_bw() +
  theme(legend.position='none') + geom_smooth(aes(epoch,train_loss), method='loess', col='black')

f2 <- ggplot(Table) + geom_line(aes(epoch,  val_loss,group=crossval,col=crossval), alpha=0.5) + theme_bw() +
  theme(legend.position='none') + geom_smooth(aes(epoch,val_loss), method='loess', col='black')

f3 <- ggplot(Table) + geom_line(aes(epoch,train_acc,group=crossval,col=crossval), alpha=0.5) + theme_bw() +
  theme(legend.position='none') + geom_smooth(aes(epoch,train_acc), method='loess', col='black')

f4 <- ggplot(Table) + geom_line(aes(epoch,  val_acc,group=crossval,col=crossval), alpha=0.5) + theme_bw() +
  theme(legend.position='none') + geom_smooth(aes(epoch,val_acc), method='loess', col='black')

pdf(file = paste(fname,'.pdf',sep=''), width=10, height=10)
  grid.arrange(f1,f2,f3,f4, ncol=2)
dev.off()

File: accanalysis.py
import pickle
import numpy as np
import pandas as pd

fname = '/Users/mviana/Desktop/deep-learning-models/log/20170626T103649'

with open(fname+'.pkl','rb') as pf:
	Data = pickle.load(pf)

FullTable = pd.DataFrame()
for crossval in range(len(Data)):
	
	subData = Data[crossval]

	nepochs = len(subData['acc'])

	trn_acc = subData['acc']
	val_acc = subData['val_acc']
	trn_los = subData['loss']
	val_los = subData['val_loss']
	indexes = np.repeat(crossval,nepochs)[0]
	epochid = np.arange(nepochs)

	Table = pd.DataFrame({'crossval':  indexes,
						  'epoch':     epochid,
		                  'train_acc': trn_acc,
		                  'val_acc':   val_acc,
		                  'train_loss':trn_los,
		                  'val_loss':  val_los})

	FullTable = FullTable.append(Table)

print(FullTable)

FullTable.to_csv(fname+'.csv', index=False)